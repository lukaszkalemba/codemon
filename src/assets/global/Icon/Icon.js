import React from 'react';
import PropTypes from 'prop-types';
import SVG from 'react-inlinesvg';

const Icon = ({ src, ...props }) => {
  return (
    <div {...props}>
      <SVG src={src} />
    </div>
  );
};

Icon.propTypes = {
  src: PropTypes.string.isRequired,
};

export default Icon;
