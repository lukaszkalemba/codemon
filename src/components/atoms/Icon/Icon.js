import React from 'react';
import PropTypes from 'prop-types';
import SVG from 'react-inlinesvg';
import S from './Icon.styles';

const Icon = ({ src, ...props }) => {
  return (
    <S.Wrapper {...props}>
      <SVG src={src} />
    </S.Wrapper>
  );
};

Icon.propTypes = {
  src: PropTypes.string.isRequired,
};

export default Icon;
