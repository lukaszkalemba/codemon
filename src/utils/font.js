export default {
  size: {
    xxxs: '1rem',
    xxs: '1.3rem',
    xs: '1.6rem',
    s: '2.2rem',
    m: '3rem',
    l: '4rem',
    xl: '5.4rem',
    xxl: '8.5rem',
    xxxl: '10rem',
  },

  weight: {
    regular: '400',
    medium: '500',
    bold: '700',
  },

  family: {
    montserrat: 'Montserrat',
  },
};
