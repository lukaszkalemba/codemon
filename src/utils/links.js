export default [
  {
    id: 0,
    path: '/projekty',
    name: 'projekty',
  },
  {
    id: 1,
    path: '/blog',
    name: 'blog',
  },
  {
    id: 2,
    path: '/wycena',
    name: 'wycena',
  },
];
