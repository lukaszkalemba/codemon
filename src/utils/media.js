export default {
  phone: '576px',
  tablet: '768px',
  desktop: '1150px',
  largeDesktop: '1600px',
};
