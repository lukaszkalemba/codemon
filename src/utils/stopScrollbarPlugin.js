import { ScrollbarPlugin } from 'smooth-scrollbar';

class stopScrollbarPlugin extends ScrollbarPlugin {
  static pluginName = 'stopScrollbar';

  static defaultOptions = {
    open: false,
  };

  transformDelta(delta) {
    return this.options.open ? { x: 0, y: 0 } : delta;
  }
}

export default stopScrollbarPlugin;
