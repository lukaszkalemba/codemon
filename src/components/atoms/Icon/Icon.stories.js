import React from 'react';
import { select, withKnobs } from '@storybook/addon-knobs';
import instagram from 'assets/svgs/icon_instagram.svg';
import facebook from 'assets/svgs/icon_facebook.svg';
import linkedin from 'assets/svgs/icon_linkedin.svg';
import dribbble from 'assets/svgs/icon_dribbble.svg';
import codemonLogo from 'assets/svgs/logo_codemon.svg';
import Icon from './Icon';

export const icon = () => (
  <div style={{ maxWidth: '200px' }}>
    <Icon
      src={select(
        'Icon Logo',
        { instagram, facebook, linkedin, dribbble, codemonLogo },
        instagram
      )}
    />
  </div>
);

export default {
  component: Icon,
  decorators: [withKnobs],
  title: 'Atoms|Icon',
};
