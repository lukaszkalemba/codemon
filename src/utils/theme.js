import color from 'utils/color';
import font from 'utils/font';
import media from 'utils/media';
import zindex from 'utils/zindex';

export default {
  color,
  font,
  media,
  zindex,
};
