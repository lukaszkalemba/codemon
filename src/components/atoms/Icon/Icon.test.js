import React from 'react';
import { render } from 'utils/test-utils';
import Icon from './Icon';

describe('<Icon />', () => {
  it('matches snapshot', () => {
    const svhPath =
      'data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjUiIGhlaWdodD0iMTYiIHZpZXdCb3g9IjAgMCA2NSAxNiIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4NCjxwYXRoIGQ9Ik0wIDFINjFMNDIuNSAxNC41IiBzdHJva2U9IndoaXRlIiBzdHJva2Utd2lkdGg9IjIiLz4NCjwvc3ZnPg0K';

    const { container } = render(<Icon src={svhPath} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
