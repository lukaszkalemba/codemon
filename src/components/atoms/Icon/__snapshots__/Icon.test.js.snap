// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> renders correctly and matches snapshot 1`] = `
<div
  class="sc-bdnylx levWdn"
>
  <svg
    fill="none"
    height="16"
    viewBox="0 0 65 16"
    width="65"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 1H61L42.5 14.5"
      stroke="white"
      stroke-width="2"
    />
  </svg>
</div>
`;
